ydb/core/blobstorage/dsproxy/dsproxy_blackboard.cpp

```diff
@@ -542,4 +542,35 @@ void TBlackboard::InvalidatePartStates(ui32 orderNumber) {
     }
 }
 
+void TBlackboard::MarkSlowDisks(TBlobState& state, bool isPut, const TAccelerationParams& accelerationParams) {
+    // by default all disks are considered fast
+    for (TBlobState::TDisk& disk : state.Disks) {
+        disk.IsSlow = false;
+    }
+
+    ui32 maxNumSlow = accelerationParams.MaxNumOfSlowDisks;
+    if (Info->GetTotalVDisksNum() <= maxNumSlow) {
+        // all disks cannot be slow
+        return;
+    }
+    
+    TDiskDelayPredictions worstDisks;
+    state.GetWorstPredictedDelaysNs(*Info, *GroupQueues,
```

Comment:
Aaand this code does partial sort for only 3 disks, right? So this won't work in general case.